grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -478,4 +502,56 @@ private void notifyExecutor() {
             executor.execute(NO_OP);
         }
     }
+
+    private boolean needsDirectInvocation() {
+        return (endpointInitialized || ctx.whenInitialized().isDone()) && ctx.eventLoop().inEventLoop();
+    }
+
+    private void execute(Runnable task) {
+        if (endpointInitialized || ctx.whenInitialized().isDone()) {
```

Comment:
We could use `assert` for this condition if we always guard this method with `needsDirectInvocation()`.

Reply:
`endpointInitialized` could be either `true`, `false` here. 
Because `endpointInitialized || ctx.whenInitialized().isDone()` and `ctx.eventLoop().inEventLoop()` are connected by `&&` condition.

Reply:
Ah, indeed. Sorry for a bad comment :sweat_smile: